src/rules/oneLineRule.ts

```diff
@@ -27,80 +25,219 @@ export class Rule extends Lint.Rules.AbstractRule {
+        this.handleFunctionLikeDeclaration(node);
+        super.visitMethodDeclaration(node);
+    }
+
+    public visitConstructorDeclaration(node: ts.ConstructorDeclaration) {
+        this.handleFunctionLikeDeclaration(node);
+        super.visitConstructorDeclaration(node);
+    }
+
+    public visitArrowFunction(node: ts.FunctionLikeDeclaration) {
+        var body = node.body;
+        if (body != null && body.kind === ts.SyntaxKind.Block) {
+            var arrowToken = OneLineWalker.getFirstChildOfKind(node, ts.SyntaxKind.EqualsGreaterThanToken);
+            var openBraceToken = node.body.getChildAt(0);
+            this.handleOpeningBrace(arrowToken, openBraceToken);
+        }
+        super.visitArrowFunction(node);
+    }
+
+    private handleFunctionLikeDeclaration(node: ts.FunctionLikeDeclaration) {
+        var body = node.body;
+        if (body != null && body.kind === ts.SyntaxKind.Block) {
+            var closeParenToken = OneLineWalker.getFirstChildOfKind(node, ts.SyntaxKind.CloseParenToken);
+            var openBraceToken = node.body.getChildAt(0);
+            this.handleOpeningBrace(closeParenToken, openBraceToken);
         }
+    }
+
+    private handleIterationStatement(node: ts.IterationStatement) {
+        // last child is the statement, second to last child is the close paren
+        var closeParenToken = node.getChildAt(node.getChildCount() - 2);
+        var statement = node.statement;
+        if (statement.kind === ts.SyntaxKind.Block) {
+            var openBraceToken = statement.getChildAt(0);
+            this.handleOpeningBrace(closeParenToken, openBraceToken);
+        }
+    }
+
+    private handleOpeningBrace(previousNode: ts.Node, openBraceToken: ts.Node) {
+        if (previousNode == null || openBraceToken == null) {
+            return;
+        }
+        var sourceFile = previousNode.getSourceFile();
+        var previousNodeLine = sourceFile.getLineAndCharacterFromPosition(previousNode.getEnd()).line;
+        var openBraceLine = sourceFile.getLineAndCharacterFromPosition(openBraceToken.getStart()).line;
+        var failure: Lint.RuleFailure;
+
+        if (this.hasOption(OPTION_BRACE) && previousNodeLine !== openBraceLine) {
+            failure = this.createFailure(openBraceToken.getStart(), openBraceToken.getWidth(), Rule.BRACE_FAILURE_STRING);
+        } else if (this.hasOption(OPTION_WHITESPACE) && previousNode.getEnd() === openBraceToken.getStart()) {
+            failure = this.createFailure(openBraceToken.getStart(), openBraceToken.getWidth(), Rule.WHITESPACE_FAILURE_STRING);
+        }
+        if (failure) {
+            this.addFailure(failure);
+        }
+    }
 
-        var kind = trivia.syntaxTriviaAt(0).kind();
-        return (kind === TypeScript.SyntaxKind.WhitespaceTrivia);
+    private static getFirstChildOfKind(node: ts.Node, kind: ts.SyntaxKind) {
+        return node.getChildren().filter((child) => child.kind === kind)[0];
```

Comment:
No it won't; I put safeguards in handleOpeningBrace in case we get an undefined node back.